File: flutter/lib/domain/use_case/deck/delete_deck_use_case.dart
import '../../entities/deck.dart';
import '../../repository/deck_repository.dart';

class DeleteDeckUseCase {
  final DeckRepository _deckRepository;

  DeleteDeckUseCase(this._deckRepository);

  Future<void> call(Deck deck) async {
    await _deckRepository.deleteDeck(deck);
  }
}

File: android/app/src/main/java/com/synergyboat/flashcardAi/domain/usecase/deck/DeleteDeckUseCase.kt
package com.synergyboat.flashcardAi.domain.usecase.deck

import com.synergyboat.flashcardAi.domain.entities.Deck
import com.synergyboat.flashcardAi.domain.repository.DeckRepository

class DeleteDeckUseCase(
    private val repository: DeckRepository
) {
    suspend operator fun invoke(deck: Deck) {
        repository.deleteDeck(deck)
    }
}